lib/meilisearch/rails/multi_search/result.rb

```diff
@@ -0,0 +1,84 @@
+module MeiliSearch
+  module Rails
+    class MultiSearchResult
+      attr_reader :metadata
+
+      def initialize(searches, raw_results)
+        @results = {}
+        @metadata = {}
+
+        searches.zip(raw_results).each do |(index_target, search_options), result|
+          index_target = search_options[:class_name].constantize if search_options[:class_name]
+
+          @results[index_target] = case index_target
+                                   when String, Symbol
+                                     result['hits']
+                                   else
+                                     load_results(index_target, result)
+                                   end
+
+          @metadata[index_target] = result.except('hits')
+        end
+      end
+
+      include Enumerable
+
+      def each_hit(&block)
+        @results.each do |_index_target, results|
+          results.each(&block)
+        end
+      end
+      alias each each_hit
+
+      def each_result
+        @results.each
+      end
+
+      def to_a
+        @results.values.flatten(1)
+      end
+      alias to_ary to_a
+
+      def to_h
+        @results
+      end
+      alias to_hash to_h
+
+      private
+
+      def load_results(klass, result)
```

Comment:
Need some guidance on how pagination would be expected to work in multi search.
Paginate the results of each search, paginate the search as a whole, or both? Waiting on a review before going through with a solution.

Reply:
As far as I know and per the docs: https://www.meilisearch.com/docs/reference/api/multi_search

You handle the pagination by each "block" of returned data. Because essentially the multi_search is just a way of making multiple requests at once.

Reply:
That checks out, since each search can have its own pagination options. This would mean that displaying all of the results together would not make much sense, since they would never be on the same "page". 

Perhaps I should expect that `#each_result` should be the main way to use a `MultiSearchResult`? It might make most sense in the documentation.

Reply:
To be honest, I am quite unsure about how the users are going to make use of this feature, so I would prefer to have some external output before applying more changes to it. But I'm on your side to whichever option you may prefer.